packages/mcdoc/src/runtime/checker/index.ts

```diff
@@ -874,7 +877,13 @@ function simplifyIndexed<T>(
 	typeDef: IndexedType,
 	context: SimplifyContext<T>,
 ): SimplifiedMcdocType {
-	const child = simplify(typeDef.child, { ...context, typeArgs: [] })
+	const child = simplify(typeDef.child, {
+		...context,
+		typeArgs: [],
+		spreadingUnion: false,
+		literalStructFields: undefined,
+		complexStructFields: undefined,
```

Comment:
Is this the only place where you'd want to reset these context variables?

Reply:
That was an obvious one, as we don't want to add the fields of a type we index into added to the parent struct.

I can't think of another case where it would be required, but I could be missing something.

Reply:
With the new approach, I am also doing this now when resolving a union member.